cli/defenseclaw/commands/cmd_setup.py

```
@@ -12450,3 +12450,115 @@ def _show_splunk_credentials(data_dir: str) -> None:
+        click.echo()
+        click.echo("  Setting up semantic model routing...")
+        click.echo()
+
+        # 1. Check/install vllm-sr
+        if shutil.which("vllm-sr"):
+            ver = _sp.run(["vllm-sr", "--version"], capture_output=True, text=True).stdout.strip()
+            click.echo(f"  ✓ vllm-sr already installed ({ver})")
+        else:
+            click.echo("  Installing vllm-sr...")
+            pkg = f"vllm-sr=={app.cfg.routing.version}" if app.cfg.routing.version else "vllm-sr"
+            result = _sp.run(["pip", "install", pkg], capture_output=True, text=True)
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)